tests/test_modes.py

```diff
@@ -100,11 +95,50 @@ def test_should_list_branch_changed_files_as_affected_tests(self):
             mode = Branch(test_file_convention)
             files, folders = mode.affected_tests()
 
-        expected_files = [
-            "tests/test_pytest_picked.py",
-            "tests/test_other_module.py",
-        ]
+        expected_files = ["tests/test_pytest_picked.py", "tests/test_other_module.py"]
         expected_folders = []
 
         assert files == expected_files
         assert folders == expected_folders
+
+    def test_should_list_changed_files(self, testdir):
+        gitroot = testdir.mkdir("gitroot")
+        gitroot.chdir()
+        try:
+            assert testdir.run("git", "init").ret == 0
+        except FileNotFoundError:
+            pytest.skip("required executable 'git' not found")
+
+        assert testdir.run("git", "config", "user.email", "_").ret == 0
+        assert testdir.run("git", "config", "user.name", "_").ret == 0
+        assert testdir.run("git", "commit", "--allow-empty", "--message=first", "--author=X<>").ret == 0
+
+        gitroot.join("test_gitroot").new(ext="py").write(b"", "wb")
+        assert testdir.run("git", "add", ".").ret == 0
+
+        output = list(Branch([]).git_output().splitlines())
+        assert output == ["test_gitroot.py"]
+
+    def test_should_only_list_changed_files_under_pytest_root(self, testdir):
+        gitroot = testdir.mkdir("gitroot")
+        gitroot.chdir()
+        try:
+            assert testdir.run("git", "init").ret == 0
+        except FileNotFoundError:
+            pytest.skip("required executable 'git' not found")
+
+        assert testdir.run("git", "config", "user.email", "_").ret == 0
+        assert testdir.run("git", "config", "user.name", "_").ret == 0
+        assert testdir.run("git", "commit", "--allow-empty", "--message=first", "--author=X<>").ret == 0
+
+        gitroot.join("test_gitroot").new(ext="py").write(b"", "wb")
+        pytestroot = gitroot.mkdir("pytestroot")
+        pytestroot.join("test_pytestroot").new(ext="py").write(b"", "wb")
+        assert testdir.run("git", "add", ".").ret == 0
+
+        output = set(Branch([]).git_output().splitlines())
```

Comment:
Is there a reason for using `list` on the previous test and `set` in this one? 😅 Sorry, I couldn't test your branch locally because git gets confused with your branch name.

Reply:
because i forgot to use `set` in both places :confounded: will fix